django/utils/jwt.py

```diff
@@ -0,0 +1,51 @@
+import json
+from datetime import datetime
+
+from django.conf import settings
+from django.core.signing import Signer, b64_decode, b64_encode
+from django.utils import timezone
+from django.utils.encoding import force_bytes
+
+
+class ExpiredToken(Exception):
+    pass
+
+
+class JWTFactory:
+    """
+    Symmetric signed JWT creation and decoding.
+    For encryption and other more advanced usages, you can use third-party
+    packages like PyJWT.
+    """
+    header = {
+        "alg": "HS256",
+        "typ": "JWT",
+    }
+    salt = None
+
+    def __init__(self, secret=None):
+        self.secret = secret or settings.SECRET_KEY
+        salt = self.salt or f'{self.__class__.__module__}.{self.__class__.__name__}'
+        self.signer = Signer(key=self.secret, salt=salt, sep='.', algorithm='sha256')
+
+    def make_token(self, payload, valid_for=None):
```

Comment:
As django 3.0 is only compatible Python ≥ 3.7, maybe we could start having some type hints?

```suggestion
    def make_token(self, payload: dict, valid_for=None):
```

Same for `valid_for`, would be nice if we had a direct hint that is a timedelta!

Reply:
Type hints are not coming to Django (yet) https://groups.google.com/d/topic/django-developers/C_Phs05kL1Q/discussion